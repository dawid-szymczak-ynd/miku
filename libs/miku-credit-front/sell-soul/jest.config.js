module.exports = {
  name: 'miku-credit-front-sell-soul',
  preset: '../../../jest.config.js',
  coverageDirectory: '../../../coverage/libs/miku-credit-front/sell-soul',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
