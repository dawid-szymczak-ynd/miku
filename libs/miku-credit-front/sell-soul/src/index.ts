export * from './lib/miku-credit-front-sell-soul.module';
