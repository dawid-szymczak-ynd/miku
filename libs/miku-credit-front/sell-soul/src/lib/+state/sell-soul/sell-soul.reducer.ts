import { RouterReducerState } from '@ngrx/router-store';
import { Action, createReducer } from '@ngrx/store';

export const SELLSOUL_FEATURE_KEY = 'sellSoul';

// tslint:disable-next-line:no-empty-interface
export interface State {}

export interface SellSoulPartialState {
  readonly [SELLSOUL_FEATURE_KEY]: State;
  readonly router: RouterReducerState;
}

export const initialState: State = {};

const sellSoulReducer = createReducer(initialState);

export function reducer(state: State | undefined, action: Action): State {
  return sellSoulReducer(state, action);
}
