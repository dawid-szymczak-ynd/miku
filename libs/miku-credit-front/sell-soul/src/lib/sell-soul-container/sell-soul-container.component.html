<p>sell-soul-container works!</p>
