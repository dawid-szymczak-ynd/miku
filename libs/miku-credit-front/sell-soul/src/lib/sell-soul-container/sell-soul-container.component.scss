:host {
  .container {
    display: grid;
    grid-template-columns: 20% 80%;
    grid-template-rows: 1fr;
    height: 100%;

    @media (max-width: 1024px) {
      grid-template-columns: 100%;
      grid-template-rows: 50px 100%;
    }

    &__left-side {
      padding: 24px;
      grid-column: 1 / 2;
      grid-row: 1 / 2;
      background-color: #fff1ed;
      color: #374b73;
      display: flex;
      align-items: flex-start;
      justify-content: center;
      text-align: center;

      @media (max-width: 1024px) {
        grid-column: 1 / 2;
        grid-row: 1 / 2;
        justify-content: left;
        align-items: center;
      }

      &__logo {
        font-size: 64px;
        height: 64px;
        width: auto;
        color: #ff8197;
        display: block;

        @media (max-width: 1024px) {
          font-size: 36px;
          height: 36px;
        }
      }

      &__title {
        font-size: 24px;
        margin: 0;

        @media (max-width: 1024px) {
          display: none;
        }
      }

      &__subtitle {
        margin: 0;
        font-size: 12px;
        line-height: 12px;
        font-weight: lighter;

        @media (max-width: 1024px) {
          display: none;
        }
      }

      &__welcome {
        margin-top: 16px;

        @media (max-width: 1024px) {
          display: none;
        }
      }
    }

    &__right-side {
      display: grid;
      grid-template-columns: 1fr 480px 1fr;
      grid-template-rows: 20% 1fr 20%;
      padding: 24px;
      grid-column: 2 / 3;
      grid-row: 1 / 2;
      background-color: #ff8197;
      color: #162d3d;

      @media (max-width: 1024px) {
        grid-template-columns: 20px 1fr 20px;
        grid-column: 1 / 2;
        grid-row: 2 / 3;
      }

      &__body {
        grid-column: 1 / 4;
        grid-row: 1 / 4;
        display: flex;
        align-items: center;

        &__title {
          text-align: center;
        }

        .loans-container {
          display: flex;
          height: 100%;
          width: 100%;

          &__item {
            background-color: #fff1ed;
            width: calc(50% - 20px);
            height: calc(50% - 20px);
          }
        }

        .form-container {
          background-color: #fff1ed;
          display: flex;
          flex-direction: column;
          padding: 32px;
          height: 90%;
          width: 100%;

          .mat-form-field {
            display: block;
          }
        }
      }
    }
  }
}
