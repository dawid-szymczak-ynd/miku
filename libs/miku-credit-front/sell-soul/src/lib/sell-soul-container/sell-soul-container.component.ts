import { Component } from '@angular/core';

@Component({
  selector: 'miku-credit-sell-soul-container',
  templateUrl: './sell-soul-container.component.html',
  styleUrls: ['./sell-soul-container.component.scss'],
})
export class SellSoulContainerComponent {
  constructor() {}
}
