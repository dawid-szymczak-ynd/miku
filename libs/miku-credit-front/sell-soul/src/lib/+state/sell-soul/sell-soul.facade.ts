import { Injectable } from '@angular/core';
import {
  CalculatePaybackPlanBody,
  LoanInterface,
  PaybackPlanInterface,
  UserInterface,
} from '@miku-credit/api-interfaces';
import { LoanFacade } from '@miku-credit/miku-credit-front/loan';
import { PaybackFacade } from '@miku-credit/miku-credit-front/payback';
import { UserFacade } from '@miku-credit/miku-credit-front/user';
import { select, Store } from '@ngrx/store';

import { Observable } from 'rxjs';

import * as fromSellSoul from './sell-soul.reducer';
import * as SellSoulSelectors from './sell-soul.selectors';

@Injectable()
export class SellSoulFacade {
  public readonly currentStep$: Observable<string> = this.store.pipe(select(SellSoulSelectors.getActiveStepIndex));
  public readonly user$: Observable<UserInterface> = this.userFacade.selectedUser$;
  public readonly loans$: Observable<LoanInterface[]> = this.loanFacade.allLoans$;
  public readonly selectedLoan$: Observable<LoanInterface> = this.loanFacade.selectedLoan$;
  public readonly paybackPlan$: Observable<PaybackPlanInterface> = this.paybackFacade.selectedPaybackPlan$;

  constructor(
    private readonly store: Store<fromSellSoul.SellSoulPartialState>,
    private readonly userFacade: UserFacade,
    private readonly loanFacade: LoanFacade,
    private readonly paybackFacade: PaybackFacade
  ) {}

  public loadLoans(page: number): void {
    this.loanFacade.loadLoansPage(page);
  }

  public selectLoan(loanId: number): void {
    this.loanFacade.selectLoan(loanId);
  }

  public calculatePaybackPlan(calculatePaybackPlanBody: CalculatePaybackPlanBody): void {
    this.paybackFacade.calculatePaybackPlan(calculatePaybackPlanBody);
  }
}
