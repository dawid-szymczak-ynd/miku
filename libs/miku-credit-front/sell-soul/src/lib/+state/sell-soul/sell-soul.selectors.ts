import { RouterReducerState } from '@ngrx/router-store';
import { createSelector } from '@ngrx/store';

import { SellSoulPartialState } from './sell-soul.reducer';

export const getRouterState = (state: SellSoulPartialState) => state.router;

export const getActiveStepIndex = createSelector(getRouterState, ({ state }: RouterReducerState) =>
  String(state.root.firstChild.firstChild.params.status)
);
