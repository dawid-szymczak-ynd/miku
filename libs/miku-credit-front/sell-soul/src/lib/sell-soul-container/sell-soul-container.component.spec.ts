import { MikuCreditFrontSellSoulModule } from '@miku-credit/miku-credit-front/sell-soul';

import { Shallow } from 'shallow-render';

import { SellSoulContainerComponent } from './sell-soul-container.component';

describe('SellSoulContainerComponent', () => {
  let shallow: Shallow<SellSoulContainerComponent>;

  beforeEach(() => {
    shallow = new Shallow(SellSoulContainerComponent, MikuCreditFrontSellSoulModule);
  });

  it('should match snapshot', async () => {
    const { fixture } = await shallow.render();

    expect(fixture).toMatchSnapshot();
  });
});
