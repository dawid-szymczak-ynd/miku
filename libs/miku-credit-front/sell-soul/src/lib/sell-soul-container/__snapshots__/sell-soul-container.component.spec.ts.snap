// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellSoulContainerComponent should match snapshot 1`] = `
<ng-component>
  <miku-credit-sell-soul-container>
    <p>
      sell-soul-container works!
    </p>
  </miku-credit-sell-soul-container>
</ng-component>
`;
